gapir/cc/context_test.cpp

```diff
@@ -46,79 +45,58 @@ class ContextTest : public ::testing::Test {
         std::vector<uint32_t> memorySizes = {MEMORY_SIZE};
         mMemoryManager.reset(new MemoryManager(memorySizes));
         mResourceProvider.reset(new StrictMock<MockResourceProvider>());
+        mConn.reset(new MockReplayConnection());
     }
 
     std::unique_ptr<MemoryManager> mMemoryManager;
     std::unique_ptr<StrictMock<MockResourceProvider>> mResourceProvider;
+    std::unique_ptr<MockReplayConnection> mConn;
 };
-
-// expectLoadReplay verifies that the a request is made for the replay with id REPLAY_ID.
-void expectLoadReplay(MockResourceProvider* resourceProvider,
-                      const std::vector<uint8_t>& replayData) {
-    Resource res(REPLAY_ID, replayData.size());
-    EXPECT_CALL(*resourceProvider, get(_, _, _, _, replayData.size()))
-            .With(Args<0, 1>(ElementsAre(res)))
-            .WillOnce(DoAll(WithArg<3>(SetVoidPointee(replayData)), Return(true)));
-}
-
 }  // anonymous namespace
 
 TEST_F(ContextTest, Create) {
-    auto replayData = createReplayData(0, 0, {}, {}, {});
-
-    expectLoadReplay(mResourceProvider.get(), replayData);
-    auto server = createServerConnection(REPLAY_ID, replayData.size());
+    auto payload = createPayload(0, 0, {}, {}, {});
+    
+    EXPECT_CALL(*mConn, getPayload()).WillOnce(Return(ByMove(std::move(payload))));
     core::CrashHandler crash_handler;
-    auto context = Context::create(*server, crash_handler, mResourceProvider.get(), mMemoryManager.get());
+    auto context = Context::create(mConn.get(), crash_handler, mResourceProvider.get(), mMemoryManager.get());
 
     EXPECT_THAT(context, NotNull());
 }
 
 TEST_F(ContextTest, CreateErrorReplayRequest) {
     // Failed to load
-    Resource res(REPLAY_ID, 10);
-    EXPECT_CALL(*mResourceProvider, get(_, _, _, _, 10))
-            .With(Args<0, 1>(ElementsAre(res)))
-            .WillOnce(Return(false));
+    EXPECT_CALL(*mConn, getPayload()).WillOnce(Return(ByMove(nullptr)));
```

Comment:
ByMove(nullptr))

Is that right? It doesn't seem right, since you cannot move a nullptr.

Reply:
It is quite interesting here. gmock, without `ByMove()` inside of `Return()`, tries to make a copy when returning the value to caller (not sure if this is due to compiler). Because `getPayload()` returns `unique_ptr`, compilation will fail due to deleted copy-ctor of `unique_ptr`.

I've changed this to `Return(ByMove(std::unique_ptr<...>(nullptr)))`. This should make more senses.